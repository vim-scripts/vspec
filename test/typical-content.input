function! s:describe__matchers__with_literals()  "{{{1
  It should succeed

  Should 0 != 1
  "Should 0 !~ '1'
  Should 0 < 1
  Should 0 <= 0
  Should 0 <= 1
  Should 0 == 0
  "Should 0 =~ '0'
  Should 0 > -1
  Should 0 >= 0
  Should 0 >= -1
  Should 0 is 0
  Should 0 isnot 1
  Should 0 isnot '0'

  Should 'a' ==# 'a'
  Should 'a' !=# 'A'
  Should 'a' !=# 'B'
  Should 'a' ==? 'A'
  Should 'a' ==? 'A'
  Should 'a' !=? 'B'

  Should [] == []
  Should [] isnot []

  Should {} == {}
  Should {} isnot {}

  Should function('function') == function('function')
  Should function('function') != function('type')

  It should fail

  Should 0 != 0
  Should 0 !~ '0'
  Should 0 < 0
  Should 0 < -1
  Should 0 <= -1
  Should 0 == 1
  Should 0 =~ '1'
  Should 0 > 0
  Should 0 > 1
  Should 0 >= 1
  Should 0 is 1
  Should 0 is '0'
  Should 0 isnot 0

  Should 'a' ==# 'b'
  Should 'a' ==# 'A'
  Should 'a' !=# 'a'
  Should 'a' ==? 'b'
  Should 'a' !=? 'A'
  Should 'a' !=? 'a'

  Should [] == [0]
  Should [] is []

  Should {} == {'0': 0}
  Should {} is {}

  Should function('function') == function('type')
  Should function('function') != function('function')
endfunction




function! s:describe__matchers__with_variables()  "{{{1
  It should succeed

  let s = 'A'
  Should s ==# 'A'
  Should s !=# 'a'

  let l = []
  Should l == []
  Should l is l
endfunction




function! s:describe__matchers__with_wrong_types()  "{{{1
  It should succeed

  Should 0 == 0
  Should 0 != '0'
  Should 0 != []

  It should fail

  Should 0 < '1'
  Should [] < []
  Should {} < {}
  Should function('type') < function('type')

  Should 0 =~ '0'
  Should [] =~ '[]'
  Should {} =~ '{}'
  Should function('type') =~ function('type')
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
