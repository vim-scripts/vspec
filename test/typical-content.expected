==== matchers with literals
---- It should succeed 
.......................
---- It should fail 
xxxxxxxxxxxxxxxxxxxxxxxxx

==== matchers with variables
---- It should succeed 
....

==== matchers with wrong types
---- It should succeed 
...
---- It should fail 
xxxxxxxx



**** Result ****


In matchers with literals

It should fail
FAILED: 0 != 0
  expected: 0
       got: 0

It should fail
FAILED: 0 !~ '0'
       lhs: 0
       rhs: '0'

It should fail
FAILED: 0 < 0
       lhs: 0
       rhs: 0

It should fail
FAILED: 0 < -1
       lhs: 0
       rhs: -1

It should fail
FAILED: 0 <= -1
       lhs: 0
       rhs: -1

It should fail
FAILED: 0 == 1
  expected: 1
       got: 0

It should fail
FAILED: 0 =~ '1'
       lhs: 0
       rhs: '1'

It should fail
FAILED: 0 > 0
       lhs: 0
       rhs: 0

It should fail
FAILED: 0 > 1
       lhs: 0
       rhs: 1

It should fail
FAILED: 0 >= 1
       lhs: 0
       rhs: 1

It should fail
FAILED: 0 is 1
  expected: 1
       got: 0

It should fail
FAILED: 0 is '0'
  expected: '0'
       got: 0

It should fail
FAILED: 0 isnot 0
  expected: 0
       got: 0

It should fail
FAILED: 'a' ==# 'b'
  expected: 'b'
       got: 'a'

It should fail
FAILED: 'a' ==# 'A'
  expected: 'A'
       got: 'a'

It should fail
FAILED: 'a' !=# 'a'
  expected: 'a'
       got: 'a'

It should fail
FAILED: 'a' ==? 'b'
  expected: 'b'
       got: 'a'

It should fail
FAILED: 'a' !=? 'A'
  expected: 'A'
       got: 'a'

It should fail
FAILED: 'a' !=? 'a'
  expected: 'a'
       got: 'a'

It should fail
FAILED: [] == [0]
  expected: [0]
       got: []

It should fail
FAILED: [] is []
  expected: []
       got: []

It should fail
FAILED: {} == {'0': 0}
  expected: {'0': 0}
       got: {}

It should fail
FAILED: {} is {}
  expected: {}
       got: {}

It should fail
FAILED: function('function') == function('type')
  expected: function('type')
       got: function('function')

It should fail
FAILED: function('function') != function('function')
  expected: function('function')
       got: function('function')


In matchers with wrong types

It should fail
FAILED: 0 < '1'
       lhs: 0
       rhs: '1'

It should fail
FAILED: [] < []
       lhs: []
       rhs: []

It should fail
FAILED: {} < {}
       lhs: {}
       rhs: {}

It should fail
FAILED: function('type') < function('type')
       lhs: function('type')
       rhs: function('type')

It should fail
FAILED: 0 =~ '0'
       lhs: 0
       rhs: '0'

It should fail
FAILED: [] =~ '[]'
       lhs: []
       rhs: '[]'

It should fail
FAILED: {} =~ '{}'
       lhs: {}
       rhs: '{}'

It should fail
FAILED: function('type') =~ function('type')
       lhs: function('type')
       rhs: function('type')


63 examples, 33 failures
